iPetCare/Application/Services/ExaminationTypesService.cs

```diff
@@ -0,0 +1,167 @@
+        public ExaminationTypesService(IServiceProvider serviceProvider) : base(serviceProvider)
+        {
+        }
+
+        public async Task<ServiceResponse<ExaminationTypesCreateDtoResponse>> CreateExaminationTypesAsync(ExaminationTypesCreateDtoRequest dto)
+        {
+            if (await Context.ExaminationTypes.Where(x => x.Name == dto.Name).AnyAsync())
+                return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.BadRequest, "Podane badanie już istanieje");
+
+            var currentUserName = CurrentlyLoggedUserName;
+            if (currentUserName != null)
+            {
+                var currentUser = await UserManager.FindByNameAsync(currentUserName);
+                if (currentUser == null)
+                    return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+                if (currentUser != null && currentUser.Role != Role.Administrator)
+                    return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.Forbidden, "Brak uprawnień");
+            }
+
+            if (currentUserName == null)
+                return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var examinationType = new ExaminationType()
+            {
+                Name = dto.Name,
+                SpeciesId = dto.SpeciesId
+            };
+
+            Context.ExaminationTypes.Add(examinationType);
+            int result = await Context.SaveChangesAsync();
+
+            if (result > 0)
+            {
+                var responseDto = new ExaminationTypesCreateDtoResponse()
+                {
+                    Name = examinationType.Name,
+                    SpeciesId = examinationType.SpeciesId,
+                    Id = examinationType.Id
+                };
+
+                return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.OK, responseDto);
+            }
+
+            return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.BadRequest);
+        }
+
+        public async Task<ServiceResponse<ExaminationTypesGetAllDtoResponse>> GetAllExaminationTypesAsync()
+        {
+            var currentUserName = CurrentlyLoggedUserName;
+
+            if (currentUserName == null)
+                return new ServiceResponse<ExaminationTypesGetAllDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var currentUser = await UserManager.FindByNameAsync(currentUserName);
+            if (currentUser == null)
+                return new ServiceResponse<ExaminationTypesGetAllDtoResponse>(HttpStatusCode.Unauthorized, "Brak uprawnień");
+
+            var examinationType = await Context.ExaminationTypes.ToListAsync();
+
+            var dto = new ExaminationTypesGetAllDtoResponse();
+            dto.ExaminationTypes = Mapper.Map<List<ExaminationTypesDetailGetAllDtoResponse>>(examinationType);
```

Comment:
To też czysto kosmetyczne, ale możemy połączyć deklarację z inicjalizacją do jednej instrukcji.

zdaje się, że będzie to jakoś: 

var dto = new ExaminationTypesGetAllDtoResponse
{
 ExaminationTypes = Mapper.Map<List<...>>
};